src/tox/session/env_select.py

```diff
@@ -302,10 +321,14 @@ def _mark_active(self) -> None:
                 for env_info in self._defined_envs_.values():
                     if labels.intersection(env_info.env.conf["labels"]):
                         env_info.is_active = True
-            if self._state.conf.options.factors:  # if matches mark it active
+            if factors:  # if matches mark it active
                 for name, env_info in self._defined_envs_.items():
-                    if factors.issubset(set(name.split("-"))):
-                        env_info.is_active = True
+                    if env_info.is_active:
+                        continue
```

Comment:
This line seems uncovered by any test:

```
src/tox/session/env_select.py (92.3%): Missing lines 327
```

Reply:
I went back to try to add a test and realized it's not possible because you can't use factors and labels in combination. Maybe it would be valuable above (it looks to me like it would apply on the `labels` trait check) but rather than overthinking it I'm just going to remove this.